File: 0443-string-compression/0443-string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
       int count=0,idx=0; char elem;

        for(int i=0;i<chars.size();i++){
            elem = chars[i];
            count = 0;
            while(i<chars.size() && chars[i] == elem){
              i++;count++;
            }
            if(count == 1) chars[idx++] = elem;
            else{
                chars[idx++] = elem;
               string no = to_string(count);
                for(char c: no){
                  chars[idx++] = c;
                }

            }
            i--;
        }
        chars.resize(idx);
        return chars.size();

    }
};

File: 0443-string-compression/0443-string-compression.java
class Solution {
    public int compress(char[] chars) {
        if (chars.length <= 1) return chars.length;

        List<Character> ls = new ArrayList<>();
        int i = 0;

        while (i < chars.length) {
            int idx = i;
            char c = chars[i];
            int cnt = 0;

            while (idx < chars.length && chars[idx] == c) {
                idx++;
                cnt++;
            }

            ls.add(c);
           if(cnt > 1){
            String count = String.valueOf(cnt);
            for (char digit : count.toCharArray()) {
                ls.add(digit);
            }}

            i = idx;
        }

        for (int k = 0; k < ls.size(); k++) {
            chars[k] = ls.get(k);
        }

        return ls.size();
    }
}
